#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.0777096272159581 -0.9924428297634961 -0.09496548577763061 4.505755098234726
  position 31.57237493146837 2.7608589843419327 -0.7471414322657286
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  translation 0 -0.06 0
  size 3385 2585
  appearance PBRAppearance {
    baseColor 0.509804 0.262745 0.156863
    baseColorMap ImageTexture {
      url [
        "textures/snow.jpg"
      ]
    }
    roughness 1
    metalness 0
  }
}
Carpet {
  translation 21.7351 0.01 0.711682
  rotation 0 1 0 4.692820414042842e-06
}
AdvertisingBoard {
  translation -1.24462 0.06 21.8258
  rotation 0 1 0 2.87979
  name "advertising board(1)"
  frontTexture [
    "images/Dronetech.png"
  ]
  backTexture [
    "images/Technex.png"
  ]
  displayWidth 5
  poleNumber 1
  poleHeight 7
  baseRatio 3.6
}
AdvertisingBoard {
  translation 2.4 0.3 -24
  rotation 0.8628563066501258 -0.3574071270187627 -0.35740612701840735 -1.7177753071795863
  frontTexture [
    "images/AMC.png"
  ]
  backTexture [
    "images/AMC.png"
  ]
  displayWidth 10.24
  displayHeight 5.12
  poleColor 0 0.886275 0.886275
  baseRatio 4.3
}
RoadPillars {
  translation 12.8503 -0.3 -12.7001
  rotation 0 1 0 2.3562003061004253
  height 8
}
RoadPillars {
  translation -12.0496 -0.3 12.7455
  rotation 0 1 0 2.3562003061004253
  name "road pillar(1)"
  height 8
}
RoadPillars {
  translation -15 -0.3 -15
  rotation 0 1 0 -2.3561996938995744
  name "road pillar(2)"
  height 8
}
RoadPillars {
  translation 13.8289 -0.3 13.3289
  rotation 0 1 0 -2.3561950010791604
  name "road pillar(3)"
  height 8
}
Roundabout {
  translation 0 0.1 0
  borderWidth 0.2
  borderHeight 0.1
  innerRadius 3
  outerRadius 6
  startRoadsLength [
    28
  ]
  startRoadsWidth 3
  startRoadsLine [
    RoadLine {
      type "none"
    }
  ]
  appearance Asphalt {
    colorOverride 0.1 0.1 1
  }
}
Solid {
  translation 29.83 1.29 10.06
  scale 2 2 2
  children [
    Group {
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
              shininess 0
              specularColor 0.1 0.5 0.07
            }
          }
          geometry Cylinder {
            height 0.15
            radius 0.6
          }
        }
      ]
    }
  ]
  name "solid(1)"
}
Solid {
  translation -30 1.31 10
  rotation 0 1 0 0.785398
  scale 2 2 2
  children [
    Group {
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
              shininess 0
              specularColor 0.1 0.5 0.07
            }
          }
          geometry Box {
            size 1 0.15 1
          }
        }
      ]
    }
  ]
  name "solid(2)"
}
Solid {
  translation 0 1.31 -10
  rotation 0 1 0 1.8326
  scale 2 2 2
  children [
    Group {
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
              shininess 0
              specularColor 0.1 0.2 0.5
            }
          }
          geometry Box {
            size 1 0.15 1
          }
        }
      ]
    }
  ]
  name "solid(4)"
}
AdvertisingBoard {
  translation 18.0558 0.04 0.688901
  rotation 0.5771968951617173 -0.5771968951617173 0.577656895078166 2.09394
  name "advertising board(2)"
  frontTexture [
    "images/image1.png"
  ]
  backTexture [
    "images/image1.png"
  ]
  displayWidth 1
  displayHeight 1
  frameThickness 0.1
}
AdvertisingBoard {
  translation -0.0144946 1.64 -10.0288
  rotation 1 0 0 1.57
  name "advertising board(6)"
  backTexture [
    "images/image2.png"
  ]
  displayWidth 0.8
  displayHeight 0.8
  frameThickness 0.1
}
AdvertisingBoard {
  translation -29.9358 1.64 10.0541
  rotation 1 0 0 1.57
  name "advertising board(3)"
  backTexture [
    "images/image3.png"
  ]
  displayWidth 0.8
  displayHeight 0.8
  frameThickness 0.1
}
AdvertisingBoard {
  translation 29.814 1.64 10.0541
  rotation 1 0 0 1.57
  name "advertising board(4)"
  backTexture [
    "images/image4.png"
  ]
  displayWidth 0.8
  displayHeight 0.8
  frameThickness 0.1
}
AdvertisingBoard {
  translation 0.0297018 1.64 30.0088
  rotation 0.9350663322496984 -0.2505500890259746 0.250750089097039 1.63709
  name "advertising board(5)"
  backTexture [
    "images/image5.png"
  ]
  displayWidth 0.8
  displayHeight 0.8
  frameThickness 0.1
}
Solid {
  translation 0 1.29 30
  scale 2 2 2
  children [
    Group {
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
              shininess 0
              specularColor 0.1 0.2 0.5
            }
          }
          geometry Cylinder {
            height 0.15
            radius 0.6
          }
        }
      ]
    }
  ]
  name "solid(3)"
}
Ceiling {
  translation 0 7.6 0
  size 150 150
}
SolidPipe {
  radius 34
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(1)"
  height 3
  radius 35
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(2)"
  height 4
  radius 36
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(3)"
  height 5
  radius 37
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(4)"
  height 6
  radius 38
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(5)"
  height 7
  radius 39
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(6)"
  height 8
  radius 40
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(7)"
  height 9
  radius 41
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(8)"
  height 10
  radius 42
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(9)"
  height 11
  radius 43
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(10)"
  height 12
  radius 44
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(11)"
  height 13
  radius 45
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(12)"
  height 14
  radius 46
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(13)"
  height 15
  radius 47
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
SolidPipe {
  name "pipe(14)"
  height 16
  radius 48
  subdivision 100
  appearance PBRAppearance {
    baseColor 1 1 0
    roughness 0.5
    metalness 0
  }
}
Ramp30deg {
  translation 29.7321 4.26326e-14 10.0663
  rotation 0 1 0 0.7853996938995746
}
Ramp30deg {
  translation -29.9969 0 10.2868
  rotation 0 1 0 1.8326
  name "ramp 30 degrees(1)"
}
Ramp30deg {
  translation -0.112761 4.44089e-16 -9.97645
  rotation 0 1 0 0.7853996938995746
  name "ramp 30 degrees(3)"
}
Ramp30deg {
  translation -0.0583508 -1.1191e-13 30.0691
  rotation 0 -1 0 -5.307179586466759e-06
  name "ramp 30 degrees(2)"
}
CeilingLight {
  translation 2 7.6 15
}
CeilingLight {
  translation 8 7.6 0
  name "ceiling light(1)"
}
CeilingLight {
  translation -2 7.6 -13
  name "ceiling light(2)"
}
CeilingLight {
  translation -4 7.6 9
  name "ceiling light(3)"
}
CeilingLight {
  translation -11 7.6 0
  name "ceiling light(4)"
}
FloorLight {
  translation -13 0 30
}
FloorLight {
  translation 30 0 -2
  name "floor light(1)"
}
SignPole {
  translation -10.7133 -0.06 -17.5461
  rotation 0 1 0 -2.0943956132800112
  height 5.2
  radius 0.04
  signBoards [
    DirectionPanel {
      translation 0 3 0
      width 2.3
      height 1
      thickness 0.3
      textColor 0.7 0 0
      textSize 0.3
      font "Roboto"
      text "Welcome to \\n our World!"
      arrow FALSE
      left TRUE
    }
  ]
}
StoneFountain {
}
CircleArena {
  radius 34
  floorAppearance Parquetry {
    type "light strip"
  }
  subdivision 64
}
Mavic2Pro {
  translation 21.764 0.1054 0.772399
  rotation -3.3905113482625346e-09 -0.7071062811861719 -0.7071072811865696 -3.1415853071795863
  controller "my_controller"
  bodySlot [
    DistanceSensor {
      translation 0.06 0 0
      children [
        DEF shape_ Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_front"
      boundingObject USE shape_
      lookupTable [
        0 2000 0
        2 100 0
        4 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.06 -0.012 0
      rotation 0 0 -1 0.523603
      children [
        DEF shape_ Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE shape_
      lookupTable [
        0 2000 0
        2 100 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.06 0.012 0
      rotation 0 0 -1 -0.523595307179586
      children [
        DEF shape_ Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE shape_
      lookupTable [
        0 2000 0
        2 100 0
      ]
      type "laser"
    }
  ]
  cameraSlot [
    Camera {
      translation 0 -0.02 0
      rotation -0.9999999999999999 0 0 1.5707953071795862
      recognitionColors [
        0 0 0
      ]
      fieldOfView 0.5
      width 720
      height 640
      far 1
    }
    Camera {
      translation 0 0.03 -0.08
      rotation 1 0 0 7.73713e-07
      name "camera_front"
      recognitionColors [
        0 0 0
      ]
      fieldOfView 0.5
      width 720
      height 640
      far 1
    }
  ]
}
